core/src/main/java/io/questdb/griffin/engine/functions/array/DoubleArraySumFunctionFactory.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.array;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.arr.ArrayView;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.DoubleFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+
+public class DoubleArraySumFunctionFactory implements FunctionFactory {
+    private static final String FUNCTION_NAME = "array_sum";
+
+    @Override
+    public String getSignature() {
+        return FUNCTION_NAME + "(D[])";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        return new Func(args.getQuick(0));
+    }
+
+    static class Func extends DoubleFunction implements UnaryFunction, DoubleUnaryArrayAccessor {
+
+        protected final Function arrayArg;
+        protected double sum = 0d;
+
+        Func(Function arrayArg) {
+            this.arrayArg = arrayArg;
+        }
+
+        @Override
+        public void applyToElement(ArrayView view, int index) {
+            double v = view.getDouble(index);
+            if (Numbers.isFinite(v)) {
```

Comment:
This isn't consistent with our "arrays don't contain NULLs" semantics. We should treat infinities as numbers you can add. But we can skip over `NaN`s.

Reply:
fixed.